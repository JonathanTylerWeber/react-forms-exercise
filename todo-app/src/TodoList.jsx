import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

function TodoList() {
    const [todos, setTodos] = useState([]);

    function addTodo(todo) {
        setTodos(prevTodos => [...prevTodos, todo]);
    }

    function deleteTodo(index) {
        setTodos(prevTodos =>
            prevTodos.filter((_, i) => i !== index)
        );
    }

    return (
        <>
            <NewTodoForm addTodo={addTodo} />
            <div>
                {todos.map((todo, index) => (
                    <Todo
                        key={index}
                        text={todo.text}
                        deleteTodo={() => deleteTodo(index)}
                    />
                ))}
            </div>
        </>
    );
}

export default TodoList;
