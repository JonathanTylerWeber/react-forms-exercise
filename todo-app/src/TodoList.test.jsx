import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
    render(<TodoList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", async function () {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("New Todo:");
    const btn = queryByText("Add");
    expect(queryByText('wash dishes')).toBeNull();
    fireEvent.change(input, { target: { value: 'wash dishes' } });
    fireEvent.click(btn);
    await waitFor(() => expect(queryByText('Todo: wash dishes')).toBeTruthy());
})


