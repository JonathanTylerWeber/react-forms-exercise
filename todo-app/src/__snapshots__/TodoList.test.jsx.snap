// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      New Todo:
    </label>
    <input
      id="text"
      name="text"
      placeholder="text"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
  <div />
</DocumentFragment>
`;
